Adding materials (#51)
ml4sts/open-humanities: episodes/16-what-is-myst.Rmd

::::::::::::::::::::::::::::::::::::::::::::::::::

:::::::: instructor

> Today is when everything we've learned come together into on.

Reviewer: ```suggestion
Today is when everything we've learned comes together.
```

> - We've talked about GitHub
> - We've seen GitHub pages do magic
> - We got intorduced to the shell of our operating systems
> - And we've learned how to do different tricks over the shell

Reviewer: ```suggestion
- We've talked about GitHub
- We've seen GitHub pages convert markdown to HTML
- We got introduced to the shell of our operating systems
- And we've learned how to manipulate files and data in the shell
```

Let's learn about using myst on the simple repo we created on the first day.

## Starting myst

> Open your terminal and `cd` into your repo's directory

Reviewer: which repo?

Type the command `code` into your terminal 

:::::::: instructor
Quick rundown on vscode view. 

> If code doesn't work (problems with path) just open vscode manually

Reviewer: ```suggestion
If code doesn't work (problems with path) open vscode manually
```

remember the demotivating language thing from instructor training

::::::::::::::::::::::::::::::::::::::::::::::::::

Notice how you can access your terminal directly from vscode too!

In your terminal type myst:

> ```bash

Reviewer: should this be `myst init` ?

This tells you that all the files that were in your directory (README.md & index.md) were built into a website locally.

::::: callout
Locally means this content is hosted on your machine. Not published for the public to be able to access.
:::::

To view this website follow the link provided in the prompt:
`http://localhost:3000`

::::: callout
If your port `3000` was already occupied you might get a different number
:::::

### What did this do?

> Firstly we notice that the files in the directory got increaced by 1. The new file is called myst.yml as mentioned by the output from the command we ran.

Reviewer: ```suggestion
Firstly we notice that the files in the directory got increased by 1. The new file is called `myst.yml` as mentioned by the output from the command we ran.
```

If we opened the webpage using the link above we see somthing like this

> ![](fig/myst_page.png)

Reviewer: please add alt text

#### What does that tell us?
We were able to create an interactive webpage using the same markdown files and turn them into an interactive page. We used the `myst` tool to make the webpage look nicer and have more powers.

We also notice that with every change we make the contents of the `_build` folder get **rebuilt** to generate the newly updated material and settings. And we see a large number of files in there that we might want to hide from the front face of our GitHub repo.

#### What did myst need to run?
We noticed earlier that myst added a file to our directory as soon as we ran it. `myst.md`
Let's figure out why it did so by viewing the contents of it and trying to understand it.

``` bash

> cat myst.md

Reviewer: is it .md or .yml?
